board/safety.h

```diff
@@ -522,36 +523,46 @@ bool steer_torque_cmd_checks(int desired_torque, int steer_req, const SteeringLi
     violation = true;
   }
 
-  // certain safety modes set their steer request bit low for one frame at a
+  // certain safety modes set their steer request bit low for one or more frame at a
   // predefined max frequency to avoid steering faults in certain situations
   bool steer_req_mismatch = (steer_req == 0) && (desired_torque != 0);
   if (steer_req_mismatch) {
-    // no torque if request bit isn't high
+    // no torque if request bit isn't high and safety mode doesn't allow tolerance
     if (!limits.has_steer_req_tolerance) {
       violation = true;
 
     } else {
-      // disallow torque cut if not enough recent matching steer_req messages
-      if (valid_steer_req_count < limits.min_valid_request_frames) {
-        violation = true;
-      }
+      if (invalid_steer_req_count == 0) {
```

Comment:
is this really needed? seems like all three checks can just always be run, though might have to account for that in the RT interval

Reply:
We reset `valid_steer_req_count` and the rt timer every time we get an invalid frame, so we'd need to conditionally reset those once we sent `max_invalid_request_frames` frames. But then if you send less than what's allowed, it will never be reset, allowing you to oscillate back and forth.